### PR TITLE
event queue for CastVoteController

### PR DESCRIPTION
**Pull request by scottPlusPlus**
* pulled in pg-boss package  (event queue using postgres)
* added abstraction around it with mock
* implemented castVoteContoller using eventQueue

will want to add an async init method to service locator to make setup cleaner, but this works for now

### CODE REVIEW

**Comment by mikefranze** [comment]
Awesome, it also looks like pg-boss has the option to set the datetime that a job should be run after. That would help us set up timed events like when the election starts/stops, email reminders, etc.

**Review by mikefranze — APPROVED** [review]

**Comment by mikefranze** [comment]
Just needs a few conflicts to be resolved @scottPlusPlus

File: backend/src/ServiceLocator.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ import ElectionRollDB from "./Models/ElectionRolls";
 import CastVoteStore from "./Models/CastVoteStore";
 import EmailService from "./Services/Email/EmailService";
 import { IBallotStore } from "./Models/IBallotStore";
+import { IEventQueue } from "./Services/EventQueue/IEventQueue";
+import PGBossEventQueue from "./Services/EventQueue/PGBossEventQueue";
+
 import AccountService from "./Services/Account/AccountService"
+import GlobalData from "./Services/GlobalData";
```
Comment (mikefranze) [inline backend/src/ServiceLocator.ts:12]: I think Services/GlobalData is missing from the commit

Comment (scottPlusPlus) [reply backend/src/ServiceLocator.ts:12]: whoops, great catch, thanks!

File: backend/src/Services/GlobalData.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default class GlobalData {
+
+  serviceUrl:string;
+
+  constructor() {
+    this.serviceUrl = "https://star-vote.herokuapp.com/";
```
Comment (mikefranze) [inline backend/src/Services/GlobalData.ts:6]: Would it make sense for this to be an environment variable?

Comment (scottPlusPlus) [reply backend/src/Services/GlobalData.ts]: Another great call :)

**Comment by mikefranze** [comment]
Looks good, thanks!

**Pull request merged**